models/operation-message.go

```diff
@@ -0,0 +1,70 @@
+package models
+
+import "encoding/json"
+
+// OperationMessage maps AsyncAPI "operationMessage" object
+type OperationMessage struct {
+	Extensions   map[string]json.RawMessage `json:"-"`
+	SchemaFormat string                     `json:"schemaFormat,omitempty"`
+	ContentType  string                     `json:"contentType,omitempty"`
+	Headers      *Schema                    `json:"headers,omitempty"`
+	Payload      json.RawMessage            `json:"payload,omitempty"`
+	Tags         []*Tag                     `json:"tags,omitempty"`
+	Summary      string                     `json:"summary,omitempty"`
+	Name         string                     `json:"name,omitempty"`
+	Title        string                     `json:"title,omitempty"`
+	Description  string                     `json:"description,omitempty"`
+	ExternalDocs *ExternalDocs              `json:"externalDocs,omitempty"`
+	Deprecated   bool                       `json:"deprecated,omitempty"`
+	Example      json.RawMessage            `json:"example,omitempty"`
+	OneOf        []*Message                 `json:"oneOf,omitempty"`
+}
```

Comment:
This model has been done by hand, to contain the two possibilities: it can be a `Message` or an object containing the key `oneOf` which in turn contains messages.

This is an example of a case in which is possible to solve the problem with `oneOf`.